crates/nargo_cli/src/cli/codegen_verifier_cmd.rs

```diff
@@ -1,22 +1,39 @@
-use super::fs::{create_named_dir, write_to_file};
+use super::fs::{create_named_dir, program::read_program_from_file, write_to_file};
 use super::NargoConfig;
-use crate::{cli::compile_cmd::compile_circuit, constants::CONTRACT_DIR, errors::CliError};
+use crate::{
+    cli::compile_cmd::compile_circuit, constants::CONTRACT_DIR, constants::TARGET_DIR,
+    errors::CliError,
+};
 use clap::Args;
 use nargo::ops::{codegen_verifier, preprocess_program};
 use noirc_driver::CompileOptions;
 
 /// Generates a Solidity verifier smart contract for the program
 #[derive(Debug, Clone, Args)]
 pub(crate) struct CodegenVerifierCommand {
+    /// The name of the circuit build files (ACIR, proving and verification keys)
+    circuit_name: Option<String>,
+
     #[clap(flatten)]
     compile_options: CompileOptions,
 }
 
 pub(crate) fn run(args: CodegenVerifierCommand, config: NargoConfig) -> Result<(), CliError> {
     let backend = crate::backends::ConcreteBackend;
 
-    let compiled_program = compile_circuit(&backend, &config.program_dir, &args.compile_options)?;
-    let preprocessed_program = preprocess_program(&backend, compiled_program)?;
+    // TODO(blaine): Should this be a utility function?
+    let circuit_build_path = args
+        .circuit_name
+        .map(|circuit_name| config.program_dir.join(TARGET_DIR).join(circuit_name));
```

Comment:
I'd like to add a "context" struct at some point which essentially wraps the `PackageManifest` along with the package's root directory. These sorts of functions would then live as methods on there.